task/upload.go

```diff
@@ -389,9 +387,6 @@ func processCatalystCallback(tctx *TaskContext, callback *clients.CatalystCallba
 			return nil, fmt.Errorf("unknown output name=%q for output=%+v", outName, output)
 		}
 	}
-	if FlagCatalystCopiesSourceFile && videoFilePath == "" {
-		return nil, fmt.Errorf("no video file path found in catalyst output")
-	}
```

Comment:
Do you mean to only return the `FlagCatalystCopiesSourceFile` check? This was meant to be a double check that Catalyst did actually return the file path for the copied file, if we expect that from it.

Reply:
After reading the rest of the code, it seems like there is no check at all that the source file was actually generated, we just trust that it was because it was asked. Would be good to either have a confirmation from Catalyst (like an explicit output on the callback) or to check here that the file is there, instead of just returning an untested path. WDYT?

Reply:
There is a different check in a previous PR here: https://github.com/livepeer/task-runner/pull/204/files#diff-d454bfad6f152c810569d9a69d364cc443c95f99e3e4171b10c088a851c8d3e3R419-R421 so if the file exists from catalyst we use it, otherwise we make the copy here in task-runner. My plan was to have it like this initially to be backwards compatible then tidy everything up afterwards removing the `complementCatalystPipeline` function.